core/templates/dev/head/pages/exploration-player-page/services/question-player-engine.service.ts

```diff
@@ -157,12 +156,10 @@ angular.module('oppia').factory('QuestionPlayerEngineService', [
        *   - initHtml {string}, an HTML string representing the content of the
        *       first state.
        */
-      init: function(questionDicts, successCallback) {
+      init: function(Questions, successCallback) {
         answerIsBeingProcessed = false;
-        for (var i = 0; i < questionDicts.length; i++) {
-          questions.push(
-            QuestionObjectFactory.createFromBackendDict(questionDicts[i])
-          );
+        for (var i = 0; i < Questions.length; i++) {
+          questions.push(Questions[i]);
```

Comment:
Can we directly assign a copy of the Question to question (instead of lopping over it)?

Also, I would suggest you to change the param Question to question and the service variable to _question (similar to _explorationId).